### PR TITLE
C#: Remove all DB stats

### PR DESCRIPTION
**Pull request by hvitved**
This PR removes all DB stats for C#, similar to [what Swift does](https://github.com/github/codeql/blob/main/swift/ql/lib/swift.dbscheme.stats).

The motivation for this change is to avoid ever having to deal with bad joins introduced when updating DB stats. Instead, we pay the penalty of fixing bad joins due to DB stats changes once and for all on this PR.

DCA an QA both look good.

### CODE REVIEW

File: csharp/ql/lib/semmle/code/csharp/AnnotatedType.qll [hunk 1]
```diff
@@ -167,9 +172,10 @@ private module Annotations {
    */
   bindingset[i]
   Nullability getChildNullability(Nullability n, int i) {
-    if nullability_parent(_, i, getNullability(n))
-    then nullability_parent(getNullability(result), i, getNullability(n))
-    else result = n
+    result = getChildNullability0(n, i)
+    or
+    not exists(getChildNullability0(n, _)) and
```
Comment (michaelnebel) [inline csharp/ql/lib/semmle/code/csharp/AnnotatedType.qll:177]:
```suggestion
    not exists(getChildNullability0(n, i)) and
```

File: csharp/ql/lib/semmle/code/csharp/AnnotatedType.qll [hunk 2]
```diff
@@ -288,7 +299,7 @@ private newtype TAnnotatedType =
     Annotations::getNoFlagsNullability(annotations) = getTypeParameterNullability(_, type)
     or
     // All types have at least one annotated type
-    Annotations::getNoFlagsNullability(annotations) instanceof Annotations::NoNullability
+    isNoFlagsNoNullability(annotations) and exists(type)
```
Comment (michaelnebel) [inline csharp/ql/lib/semmle/code/csharp/AnnotatedType.qll:302]: Why is `exists(type)` needed?

File: csharp/ql/lib/semmle/code/csharp/controlflow/Guards.qll [hunk 3]
```diff
@@ -1111,36 +1117,47 @@ module Internal {
...
+      PreBasicBlocks::PreBasicBlock bbGuard, PreBasicBlocks::PreBasicBlock bbPhi
     ) {
       e = upd.getDefinition().getSource() and
       upd = phi.getAnInput() and
-      preControlsDirect(guard, upd.getBasicBlock(), vGuard) and
+      preControlsDefDirect(guard, upd, vGuard) and
       bbGuard.getAnElement() = guard and
-      bbGuard.strictlyDominates(phi.getBasicBlock()) and
-      not preControlsDirect(guard, phi.getBasicBlock(), vGuard)
+      bbPhi = phi.getBasicBlock()
```
Comment (michaelnebel) [inline csharp/ql/lib/semmle/code/csharp/controlflow/Guards.qll:1129]: Maybe use `instanceof` instead as `bbPhi` is used?

**Review by michaelnebel — COMMENTED** [review]
Wow - that is a lot of performance tweaks in one PR! :-)
Well done!

File: csharp/ql/lib/semmle/code/csharp/AnnotatedType.qll (hunk 1 above)
Comment (hvitved) [reply csharp/ql/lib/semmle/code/csharp/AnnotatedType.qll:177]: Ouch.

**Review by michaelnebel — APPROVED** [review]
LGTM!

**Pull request merged**